src/js/videojs-dash.js

```diff
@@ -70,6 +70,93 @@ class Html5DashJS {
     // element to bind to.
     this.mediaPlayer_.initialize();
 
+    // Retrigger a dash.js-specific error event as a player error
+    // See src/streaming/utils/ErrorHandler.js in dash.js code
+    // Handled with error (playback is stopped):
+    // - capabilityError
+    // - downloadError
+    // - manifestError
+    // - mediaSourceError
+    // - mediaKeySessionError
+    // Not handled:
+    // - timedTextError (video can still play)
+    // - mediaKeyMessageError (only fires under 'might not work' circumstances)
+    this.retriggerError_ = (event) => {
+      if (event.error === 'capability' && event.event === 'mediasource') {
+        // No support for MSE
+        this.player.error({
+          code: 4,
+          message: 'The media cannot be played because it requires a feature ' +
+            'that your browser does not support.'
+        });
+
+      } else if (event.error === 'manifestError' && (
+          (event.event.id === 'createParser') || // Manifest type not supported
+          (event.event.id === 'codec') || // Codec(s) not supported
+          (event.event.id === 'nostreams') || // No streams available to stream
+          (event.event.id === 'nostreamscomposed') || // Error creating Stream object
+          (event.event.id === 'parse') || // syntax error parsing the manifest
+          (event.event.id === 'multiplexedrep') // a stream has multiplexed audio+video
+        )) {
+        // These errors have useful error messages, so we forward it on
+        this.player.error({code: 4, message: event.event.message});
+
+      } else if (event.error === 'mediasource') {
+        // This error happens when dash.js fails to allocate a SourceBuffer
+        // OR the underlying video element throws a `MediaError`.
+        // If it's a buffer allocation fail, the message states which buffer
+        // (audio/video/text) failed allocation.
+        // If it's a `MediaError`, dash.js inspects the error object for
+        // additional information to append to the error type.
+        if (event.event.match('MEDIA_ERR_ABORTED')) {
+          this.player.error({code: 1, message: event.event});
+        } else if (event.event.match('MEDIA_ERR_NETWORK')) {
+          this.player.error({code: 2, message: event.event});
+        } else if (event.event.match('MEDIA_ERR_DECODE')) {
+          this.player.error({code: 3, message: event.event});
+        } else if (event.event.match('MEDIA_ERR_SRC_NOT_SUPPORTED')) {
+          this.player.error({code: 4, message: event.event});
+        } else if (event.event.match('MEDIA_ERR_ENCRYPTED')) {
+          this.player.error({code: 5, message: event.event});
+        } else {
+          this.player.error({code: 4, message: event.event});
```

Comment:
Any specific reason for going with `MEDIA_ERR_SRC_NOT_SUPPORTED` instead of `MEDIA_ERR_UNKNOWN` here?

Reply:
Mmm, kinda but not really. Is `MEDIA_ERR_UNKNOWN` a real thing? It's not defined in the w3c specs. I assumed it was just video.js-specific.

I only ask because this error comes directly from the `video` element, so I figured it should throw something that's within spec, and `MEDIA_ERR_SRC_NOT_SUPPORTED` seemed the most logical at the time. That being said, it really shouldn't throw anything other than the errors explicitly listed above, so I guess my point is moot.

Reply:
Yeah, something within spec is probably best, it just seemed odd given that it's a 'mediasource' error. I think sticking with it is OK, but maybe add a comment that we shouldn't get to this point, and we're defaulting to MEDIA_ERR_SRC_NOT_SUPPORTED since it seemed the most contextually accurate of the available errors.

Reply:
I misspoke; that last `else` covers both the case that an unknown error comes from the `video` element, and also when the source buffer allocations fail (as mentioned in the comments just above these lines). In the case of an unknown error coming from the `video` element, the message will always contain `UNKNOWN`, so I can match for that if you want, to split apart those two cases.

An MSE error would be considered `MEDIA_ERR_SRC_NOT_SUPPORTED`, right?

Reply:
Made separate cases for the `UNKNOWN` `<video>`-dispatched error and the can't create source buffer error. Let me know if I should change the `code` for the source buffer error.

Reply:
Can an `UNKNOWN` error come from the `video` element?

Reply:
I think you were right that https://html.spec.whatwg.org/multipage/media.html#error-codes doesn't include any `UNKNOWN`, and `MEDIA_ERR_UNKNOWN` is video.js specific. That said, if we don't want to trigger the video.js specific error, I think an error with an appropriate message regarding unknown should be fine.

Reply:
So are the latest changes good with you, then?

Reply:
I was more going back on the whole UNKNOWN piece. As in, since UNKNOWN is only video.js (correct me if I'm wrong) then `} else if (event.event.match('UNKNOWN')) {` will never happen.

Reply:
This error (`event.error === 'mediasource'`) is thrown from [here](https://github.com/Dash-Industry-Forum/dash.js/blob/development/src/streaming/controllers/StreamController.js#L619-L638), where you can see that it's making a conversion from numerical `code` to a string. In its default case, it makes that string `UNKNOWN`. If you unwind the call you end up back [here](https://github.com/Dash-Industry-Forum/dash.js/blob/development/src/streaming/controllers/PlaybackController.js#L561), where the event listener is attached to the native element.

So, from the holistic point of view, it should never happen. But, if you just black box dash.js and consider the errors it can throw as an API, then it _might_ happen.

(The `event.error === 'mediasource'` error is also thrown with just a standard "can't allocate buffer" string when [the call to `addSourceBuffer` fails](https://github.com/Dash-Industry-Forum/dash.js/blob/development/src/streaming/controllers/BufferController.js#L133). The dash.js error handling is _very_ messy.)

Reply:
Thank you, makes sense now.